src/main/java/com/artipie/asto/KeyComparator.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto;
+
+import java.util.Comparator;
+
+/**
+ * Comparator for key values by their string representation.
+ * @param <T> Type of key
+ * @since 1.1.0
+ */
+public final class KeyComparator<T extends Key> implements Comparator<T> {
```

Comment:
This class is stateless, so it could be implemented as anonymous class in `Key` static field:
```java
interface Key {
  Comparator<Key> CMP_STRING = (first, second) -> first.string().compareTo(second.string())
}
```
And it doesn't make a sense to use `T` generic which extends `Key`, the comparator interface accepts generic parameters, not provide/return it, so comparators are used as parameter with `? super T` of `Comparator<T>`, e.g. `List<T>:sort` accepts `Comparator<? super T>` to sort the list.